docs/source/platform/federation.mdx

```diff
@@ -0,0 +1,216 @@
+1. Wait for health checks to pass against the canary, watch dashboards, etc.
+1. After the canary is stable, roll out the changes to the rest of production using
+  ```
+  apollo service:push --tag=prod --serviceName=foo
+  ```
+
+Because you can [tag metrics with variants](https://www.apollographql.com/docs/platform/schema-registry/#associating-metrics-with-a-variant) as well, you can use [Apollo Graph Manager](https://engine.apollographql.com) to verify a canary's performance before rolling out changes to the rest of the graph. You can also use a similar strategy with variants to support a variety of other advanced deployment workflows, like blue/green deployments.
+
+### Controlling rollout
+
+To begin, ensure all component services are up and running, and the working directory has a `.env` file containing the appropriate `ENGINE_API_KEY` for your federated service. Then, upload each service to the gateway through `Apollo CLI`:
+
+```bash
+apollo service:push       \
+  --serviceURL={{url}}    \
+  --endpoint={{url}}      \
+  --serviceName={{name}}  \
+  --federated
+```
+
+> Each service needs a unique name, this is how you will identify the service for schema updates
+
+The given `endpoint` will be queried for the new schema which, if compatible with the rest of the schema, will be uploaded to `Apollo Graph Manager`. At this point, any gateways serving this graph will begin to roll over to the new schema.
+
+You can later modify hosted service definitions by rerunning the `apollo service:push` command with the service's name and the new URL.
+
+> What's the difference between `serviceURL` and `endpoint` parameters? The `endpoint` parameter controls the endpoint where the schema will be fetched from at composition, whereas `serviceURL` controls what URL the gateway will query at runtime. This is especially helpful if you have disabled introspection queries in your production services.
+
+// TODO: Document the integration with service:push, services under a graph, and gateway lifecycle management utilities
+
+### Reliability
+
+[//]: # (Description: This section should document how the gateway polls GCS for updates, why it's a reliable model, what the defaults are, and any recommendations)
+[//]: # (Assignee: Jackson)
+[//]: # (Reviewer: Adam)
+
+When operating as a managed gateway, `Apollo Server` will poll the [Schema Registry](https://www.apollographql.com/docs/platform/schema-registry/) for updates to the registered service list at 10 second intervals. When a service list update does occur, `Apollo Server` will create a new composed schema from the federated services, and begin to roll over to serving the new schema. Existing in-flight operations on the old schema will continue to be processed while serving the new schema. For this reason, it can be helpful to serve the new schemas from new endpoints, such that no downtime is incurred during rollover.
```

Comment:
I think we should have additional information here about the fact that this isn't polling the schema registry directly, but rather a cloud file system over a CDN. The whole point is that the reliability of our infrastructure isn't critical for the reliability of your graph.